### PR TITLE
[zh-cn]: Update translation of Promise.resolve()

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#13733

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md [hunk 1]
```diff
@@ -5,128 +5,208 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/resolve
 
 {{JSRef}}
 
-**`Promise.resolve(value)`** 方法返回一个以给定值解析后的 {{jsxref("Promise")}} 对象。如果这个值是一个 promise，那么将返回这个 promise；如果这个值是 thenable（即带有 {{jsxref("Promise.then", "then")}} 方法），返回的 promise 会“跟随”这个 thenable 的对象，采用它的最终状态；否则返回的 promise 将以此值完成。此函数将类 promise 对象的多层嵌套展平。
+**`Promise.resolve()`** 静态方法将给定的值转换为一个 {{jsxref("Promise")}}。如果该值本身就是一个 Promise，那么该 Promise 将被返回；如果该值是一个 [thenable](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenable)，`Promise.resolve()` 将调用 `then()` 方法及其两个回调函数；否则，返回的 Promise 将会以该值兑现。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md:8]:
```suggestion
**`Promise.resolve()`** 静态方法将给定的值转换为一个 {{jsxref("Promise")}}。如果该值本身就是一个 Promise，那么该 Promise 将被返回；如果该值是一个 [thenable](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenable) 对象，`Promise.resolve()` 将调用 `then()` 方法及其两个回调函数；否则，返回的 Promise 将会以该值兑现。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md [hunk 2]
```diff
@@ -5,128 +5,208 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/resolve
...
-value
-
-将被 `Promise` 对象解析的参数，也可以是一个`Promise` 对象，或者是一个 thenable。
+- `value`
+  - : 该 `Promise` 对象要被解决的参数。也可以是一个 `Promise` 对象或一个 thenable 对象。
 
 ### 返回值
 
-返回一个带着给定值解析过的 `Promise` 对象，如果参数本身就是一个 `Promise` 对象，则直接返回这个 `Promise` 对象。
+一个由给定值解决的 {{jsxref("Promise")}}，或者如果该值为一个 Promise 对象，则返回该对象。一个已解决的 Promise 可以处于已兑现、已拒绝或待定状态中的任何一种。例如，对一个已拒绝的 Promise 进行解决仍将返回一个已拒绝的 Promise。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md:27]:
```suggestion
一个由给定值解决的 {{jsxref("Promise")}}，或者如果该值为一个 Promise 对象，则返回该对象。一个已解决的 Promise 可以处于已兑现、已拒绝或待定状态中的任何一种。例如，对一个已拒绝的 Promise 进行调用仍将返回一个已拒绝的 Promise。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md [hunk 3]
```diff
@@ -5,128 +5,208 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/resolve
...
-静态方法 `Promise.resolve` 返回一个解析过的 `Promise` 对象。
+`Promise.resolve()` 方法用于*解决*（resolve）一个 Promise 对象，但这与“兑现”（fulfill）或“拒绝”（reject）Promise 对象并不相同。有关术语的定义，请参见 [Promise 描述](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#描述)。简而言之，`Promise.resolve()` 返回一个 Promise 对象，其最终状态取决于另一个 Promise 对象、thenable 对象或其他值。
+
+`Promise.resolve()` 方法是一个通用方法，支持子类化，这意味着它可以在 `Promise` 的子类上调用，并且结果会是该子类类型的 Promise 对象。为此，子类的构造函数必须实现与 [`Promise()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise/Promise) 构造函数相同的签名，即接受一个 `executor` 函数，该函数可以用 `resolve` 和 `reject` 回调函数作为参数进行调用。
+
+`Promise.resolve()` 方法特殊处理了原生 `Promise` 实例。如果 `value` 属于 `Promise` 或其子类，并且 `value.constructor === Promise`，那么 `Promise.resolve()` 直接返回 `value`，而不会创建一个新的 `Promise` 实例。否则，`Promise.resolve()` 实际上相当于 `new Promise((resolve) => resolve(value))` 的简写形式。
+
+实际上，大部分解决逻辑是由 `Promise()` 构造函数传递的 [resolver 函数](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise/Promise#resolver_函数)实现的，简单概括如下：
+
+- 如果传入的是一个非 [thenable](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenable) 值，则返回的 Promise 对象将以该值兑现。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md:39]:
```suggestion
- 如果传入的是一个非 [thenable](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise#thenable) 对象的值，则返回的 Promise 对象将以该值兑现。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md [hunk 4]
```diff
@@ -5,128 +5,208 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/resolve
...
+- 如果传入的是一个 thenable 对象，则通过传入一对解决函数作为参数调用该 thenable 对象的 `then` 方法后得到的状态将作为返回的 Promise 对象的状态。（但是因为原生的 Promise 直接通过 `Promise.resolve()` 调用，而不创建封装器，所以不会在原生 Promise 上调用 `then` 方法。）如果 resolver 函数接收到另一个 thenable 对象，则会再次进行解决，以确保 Promise 对象的最终兑现值永远不会是 thenable 对象。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md:40]:
```suggestion
- 如果传入的是一个 thenable 对象，则通过传入一对解决函数作为参数调用该 thenable 对象的 `then` 方法后得到的状态将作为返回的 Promise 对象的状态。（但是因为原生的 Promise 直接通过 `Promise.resolve()` 调用，而不创建封装对象，所以不会在原生 Promise 上调用 `then` 方法。）如果 resolver 函数接收到另一个 thenable 对象，则会再次进行解决，以确保 Promise 对象的最终兑现值永远不会是 thenable 对象。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md [hunk 5]
```diff
@@ -5,128 +5,208 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/resolve
...
-value
-
-将被 `Promise` 对象解析的参数，也可以是一个`Promise` 对象，或者是一个 thenable。
+- `value`
+  - : 该 `Promise` 对象要被解决的参数。也可以是一个 `Promise` 对象或一个 thenable 对象。
 
 ### 返回值
 
-返回一个带着给定值解析过的 `Promise` 对象，如果参数本身就是一个 `Promise` 对象，则直接返回这个 `Promise` 对象。
+一个由给定值解决的 {{jsxref("Promise")}}，或者如果该值为一个 Promise 对象，则返回该对象。一个已解决的 Promise 可以处于已兑现、已拒绝或待定状态中的任何一种。例如，对一个已拒绝的 Promise 进行调用仍将返回一个已拒绝的 Promise。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md:27]:
```suggestion
一个由给定值解决的 {{jsxref("Promise")}}，或者如果该值为一个 Promise 对象，则返回该对象。用于解决的 Promise 可以处于已兑现、已拒绝或待定状态中的任何一种。例如，对一个已拒绝的 Promise 进行调用仍将返回一个已拒绝的 Promise。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md [hunk 6]
```diff
@@ -5,128 +5,208 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/resolve
...
-*  original === cast ? true
-*  value: 33
-*/
+// 按顺序打印：
+// original === cast ? true
+// 值：33
 ```
 
-日志顺序颠倒其实是由于异步地调用 `then` 方法。在[这里](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#Return_value)查看 `then` 是如何工作的。
+日志输出顺序颠倒其实是由于异步地调用 `then` 方法。请参见 {{jsxref("Promise/then", "then()")}} 参考文档获取更多信息。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md:83]:
```suggestion
日志输出顺序颠倒其实是由于异步地调用 `then` 处理器。请参见 {{jsxref("Promise/then", "then()")}} 参考文档获取更多信息。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md [hunk 7]
```diff
@@ -5,128 +5,208 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/resolve
...
+  (v) => {
+    // 不会被调用
+  },
+  (e) => {
+    console.error(e); // TypeError: 抛出异常
+  },
+);
+
+// Thenable 在回调之后抛出异常
+// Promise 被解决
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md:125]:
```suggestion
// Thenable 在回调 Promise 被解决之后抛出异常
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md [hunk 8]
```diff
@@ -5,128 +5,208 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/resolve
...
+    });
+  },
+};
+
+Promise.resolve(thenable).then((v) => {
+  console.log(v); // 42
 });
 ```
 
+> **警告：** 不要在一个解决为自身的 thenable 对象上调用 `Promise.resolve()`。这将导致无限递归，因为它试图展平一个无限嵌套的 Promise 对象。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md:163]:
```suggestion
> **警告：** 不要在一个解决为自身的 thenable 对象上调用 `Promise.resolve()`。这将导致无限递归，因为它试图展平一个无限嵌套的 Promise。
```

**Review by yin1999 — COMMENTED** [review]
Some suggestions

File: files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md (hunk 8 above)
Comment (JasonLamv-t) [reply files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md:163]: Why delete "对象" here?

Comment (yin1999) [reply files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md:162]: I think it's not necessary to emphasis "object", couldn't we use the same expression as the English document?

https://github.com/mdn/content/blob/6de198bfa42e1f77c5f050f56da1cb345a596d45/files/en-us/web/javascript/reference/global_objects/promise/resolve/index.md?plain=1#L165

Comment (JasonLamv-t) [reply files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md:162]: Okay, I add "object" just want may it more in line with reading/oral habits.

File: files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md [hunk 9]
```diff
@@ -5,128 +5,207 @@ slug: Web/JavaScript/Reference/Global_Objects/Promise/resolve
...
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/promise/resolve/index.md:162]:
```suggestion
> **警告：** 不要在一个解决为自身的 thenable 对象上调用 `Promise.resolve()`。这将导致无限递归，因为它试图展平一个无限嵌套的 Promise。
```

**Review by yin1999 — APPROVED** [review]
Thanks

**Pull request merged**